### DISCUSSION TITLE
Draft Mode: Support Websockets for real live-preview

### DISCUSSION THREAD

**Discussion by LasseRosenow** [vercel/next.js#50642] [App Router]
It would be really helpful if with an enabled `Draft Mode`, it were possible to update the content of the webpage live via for example `Websockets`.
This allows `real live-preview` instead of needing to reload the page every time to get the new data.

## Use Case
I have `Headless-CMS` which has a `live-preview` feature that embeds the page via an iframe.
The `Headless-CMS` allows me to subscribe to a preview endpoint, which continuously returns the data I currently type into my CMS.
So when I write something on my CMS, it immediately gets pushed to the client website without me needing to save, so I can in theory see my input being rendered on the client in real-time.

But with next.js this does not work so well, because even in `Draft Mode` the server prerenders the pages content and sends a static page to the client browser.
I need some way for next.js to update that client browser without having to reload the whole page on every change.

**Comment by mehdibha** [comment]
You can use SSE instead of websockets (https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events)
